### PR TITLE
Enhancements (Docker, PyLint, PyCodeStyle, Travis-CI)

### PR DESCRIPTION
**Pull request by farhan-arshad-dev**
### Project code and feature Enhancements
- Integration of docker, travis-ci, and PyLint, PyCodeStyle.
- Resolve PyLint, PyCodeStyle reported issues.
- Added Unit tests for the project feature. 
- Using TDD approach to update the admin page for user profile model. 
- Configure Postgresql.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    # define that app service is depends on the db service this means two things
+    # 1. Database service start before the app service
+    # 2. Database service will e available via the network while using the
+    # host name 'db' and can connect with the host name and the container that
+    # is running on the 'db' service
+    depends_on: # it can be list of service 
+      - db
+
+  # service for database called 'db'  
+  db: # name of the service for the database
```
Comment (waheedahmed) [inline docker-compose.yml]: Move `db` at the top then you don't need `wait_for_db` command I believe.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+services:
+  - docker
+
+# insall docker compose before starting the travis
+before_script: pip install docker-compose
+
+script:
+  # run the docker container and then execute the test and linting tool
+  - docker-compose run app sh -c "python recipe_management_project/manage.py test && flake8"
```
Comment (waheedahmed) [inline .travis.yml]: Always add a new line at end of file, add here and in other files too where missing.

**Review by waheedahmed — CHANGES_REQUESTED** [review]

File: src/recipe_management_project/api/management/commands/wait_for_db.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# 'time' use for the application sleep for some sec
```
Comment (waheedahmed) [inline src/recipe_management_project/api/management/commands/wait_for_db.py]: Remove this command and related tests if not needed.

File: src/recipe_management_project/api/views.py [hunk 4]
```diff
@@ -103,15 +103,22 @@ def perform_create(self, serializer):
...
+        if(following_to.id == self.request.user.id):
             raise NotAcceptable("You can't follow your own profile")
 
-        # check user is trying to follow already following user 
-        if(following_models.FollowingModel.objects.filter(following_to = serializer.validated_data['following_to'].id,\
-            following_by=self.request.user.id)):
-            raise NotAcceptable("You already following {} user".format(serializer.validated_data['following_to']))
-        
+        # check user is trying to follow already following user
+        is_already_following = following_models.FollowingModel.objects.\
```
Comment (waheedahmed) [inline src/recipe_management_project/api/views.py]: nit: 
```
is_already_following = following_models.FollowingModel.objects.filter(
    following_to=following_to.id,
    following_by=self.request.user.id
)
```

File: src/recipe_management_project/api/views.py [hunk 5]
```diff
@@ -103,15 +103,22 @@ def perform_create(self, serializer):
...
+        if(is_already_following):
+            raise NotAcceptable("You already following {} user".format(
+                serializer.validated_data['following_to']))
+
         serializer.save(following_by=self.request.user)
-             
+
     def get_queryset(self):
-        return following_models.FollowingModel.objects.filter(following_by=self.request.user.id)
+        return following_models.FollowingModel.objects.\
```
Comment (waheedahmed) [inline src/recipe_management_project/api/views.py]: Same here.

**Review by waheedahmed — CHANGES_REQUESTED** [review]
Generally looks good, some small nits. Integerate Pylint and PEP8 and run them in travis build.

**Comment by farhan-arshad-dev** [comment]
> Generally looks good, some small nits. Integerate Pylint and PEP8 and run them in travis build.

@waheedahmed after Integration of `Pylint`, are we need `flake8`?

**Comment by waheedahmed** [comment]
@farhan-arshad-dev No, but do integerate PEP8 and also check `flake8` is correctly running during your build and reporting the voilations correctly if present.

**Comment by waheedahmed** [comment]
I don't see any log present in your build saying anything related to if flake8 command ran successfully or not.

**Comment by farhan-arshad-dev** [comment]
> @farhan-arshad-dev No, but do integerate PEP8 and also check `flake8` is correctly running during your build and reporting the voilations correctly if present.

flake8 is working fine, its show report if there is any error in the code.

**Comment by farhan-arshad-dev** [comment]
> > Generally looks good, some small nits. Integerate Pylint and PEP8 and run them in travis build.
> 
> @waheedahmed after Integration of `Pylint`, are we need `flake8`?

Are we need to integrate the `Pylint` and `PEP8` (renamed as `pycodestyle`) in docker container?

**Review by waheedahmed — APPROVED** [review]

**Pull request merged**